lib/spack/spack/solver/asp.py

```diff
@@ -1962,19 +1969,20 @@ def error(self, priority, msg, *args):
             msg = msg.replace('Spec(%s)' % spec_str, str(spack.spec.Spec(spec_str)))
         raise UnsatisfiableSpecError(msg)
 
-    def variant_value(self, pkg, name, value):
+    def variant_value(self, pkg, name, value, propagate=False):
```

Comment:
I don't think this `propagate` argument is actually doing anything... `variant_value` facts in the concretizer don't have a fourth field, so it will never be `True`. And it's only passed to `patches` and `dev_path` variants, which would never be propagated since they're generated by Spack internally.